lib/validators.rb

```diff
@@ -137,9 +137,107 @@ def check_shipments_specificities
       }
     end
 
+    def vehicle_days_availability(vehicles_set, timewindows_set)
+      vehicles_days = timewindows_set.collect{ |timewindow|
+        if timewindow.empty?
+          []
+        else
+          days = timewindow.flat_map{ |tw| tw[:day_index] }
+          days.compact!
+          days.uniq
+        end
+      }.delete_if(&:empty?)
+
+      vehicles_unavailable_indices = vehicles_set.collect{ |v| v[:unavailable_work_day_indices] }
+      vehicles_unavailable_indices.compact!
+
+      [vehicles_days, vehicles_unavailable_indices]
+    end
+
+    def check_vehicle_trips_stores_consistency(relation_vehicles)
+      relation_vehicles.each_with_index{ |vehicle_trip, v_i|
+        case v_i
+        when 0
+          unless vehicle_trip[:end_point_id]
+            raise OptimizerWrapper::DiscordantProblemError.new('First trip should at least have end point id')
```

Comment:
First trip should at least have **an** end point id

Cannot do suggestions :thinking:  I guess it is beacuse I am reviewing commit by commit :(

Reply:
Haha no problem. Yes it is, because it assumes that this line can change from this commit to the end of PR.

Reply:
I think you can resolve typo/grammar "type" conversations while you are making the corrections if you don't mind. There is no point for me to go over them again.